core/src/test/kotlin/org/evomaster/core/problem/rest/classifier/ModelMetricsWithTimeWindowTest.kt

```diff
@@ -0,0 +1,71 @@
+package org.evomaster.core.problem.rest.classifier
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+
+class ModelMetricsWithTimeWindowTest {
+
+
+    @Test
+    fun estimateModelMetrics() {
+
+        val delta = 0.0001
+        val ma = ModelMetricsWithTimeWindow(10) // buffer large enough to hold all
+
+        // 1) correct prediction: 200 vs. 200 -> TN
+        ma.updatePerformance(predictedStatusCode = 200, actualStatusCode = 200)
```

Comment:
keep this test. but add a new one (can copy&paste some parts of this test)in which the amount of data is more than the buffer. eg you could have a buffer of size 2, and insert the first 5 elements in this test